server/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -162,8 +164,8 @@ public BytesRef next() {
         }
 
         @Override
-        public void fillWith(StreamInput in) throws IOException {
-            in.readBytes(array, 0, Math.toIntExact(size()));
+        public void fillWith(InputStream in) throws IOException {
```

Comment:
See CCR changes, we were doing a really hacking thing there reading into a `BytesReference` via it's iterator over `BytesRef`. Just passing the `ByteArray` through the CCR code and reading via this method.